### PR TITLE
Adapt to country template training upgrade

### PR DESCRIPTION
**Pull request by bonjourmauko**
#### Technical changes

- Adapt package to country template training upgrade
  - Remove use of obsolete `pkg_resources`
  - Refactor redundant metadata handling logic

### CODE REVIEW

**Review by MattiSG — APPROVED** [review]
Looks good!

**Comment by bonjourmauko** [comment]
Please let me know @verbman how would you like to proceed here. These changes require that a `pyproject.toml` is in place.

File: openfisca_tasks/test_code.mk [hunk 1]
```diff
@@ -10,7 +10,7 @@ install: install-deps install-edit install-test
 ## Enable regression testing with template repositories.
 install-test:
 	@$(call print_help,$@:)
-	@pip install --upgrade --no-dependencies openfisca-country-template
+	@pip install git+https://github.com/openfisca/country-template@training-upgrade-fix
```
Comment (MattiSG) [inline openfisca_tasks/test_code.mk:13]: v7 [has been published](https://pypi.org/project/openfisca-country-template/) and should integrate the required fix via 49726ba24fcd61c6d1a0629077ff98d997f96132 🙂 
```suggestion
	@pip install --upgrade --no-dependencies openfisca-country-template
```

**Comment by MattiSG** [comment]
Thanks a lot @bonjourmauko! Is there any reason left for this PR to be a draft? 🙂

**Comment by MattiSG** [comment]
I take responsibility to proceed with that PR since the country template has been upgraded and Core CI tests will fail otherwise.

**Pull request merged**